tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -1037,6 +1038,34 @@ private[optimizer] abstract class OptimizerCore(
     }
   }
 
+  /** Resolves any [[RecordType]] in a [[PreTransform]]. */
+  private def resolveRecordType(
+      preTrans: PreTransform): Option[(RecordType, CancelFun)] = {
```

Comment:
No. `preTrans.tpe` would be the refined *original* type, not the record type it's been lowered to. For example, given
```scala
@inline class Foo(val x: Int)
val foo = new Foo(5)
```
`resolveRecordType(<foo's LocalDef>) == RecordType("x" -> RecordField(...))`
whereas
`preTrans.tpe == RefinedType(ClassType("Foo"), isExact = true, isNullable = false)`

That's why we have to do the same kind of resolution as `resolveLocalDef`, but we short-cut it so that we only find what we're interested in, rather than actually go and force the `PreTransform`.